### PR TITLE
SW-3275 Index a person

### PR DESCRIPTION
**Pull request by devkokov**
Person data model, validation and swagger docs

### CODE REVIEW

File: elasticsearch/client.go [hunk 1]
```diff
@@ -63,16 +73,13 @@ func (c Client) Index(i Indexable) *IndexResult {
 	req.Header.Add("Content-Type", "application/json")
 
 	// Sign the request, send it, and print the response
-	_, err = signer.Sign(req, body, service, region, time.Now())
-	if err != nil {
-		c.logger.Println(err.Error())
-	}
+	_, _ = signer.Sign(req, body, service, region, time.Now())
```
Comment (denialanderror) [inline elasticsearch/client.go:76]: Do we not care if an error occurs here? If not, you could just do `signer.Sign(req, body, service, region, time.Now())` without the assignment.

File: middleware/jwt_auth.go [hunk 2]
```diff
@@ -74,6 +74,6 @@ func verifyToken(header string, secret string) (*jwt.Token, *authorisationError)
 
 func hashEmail(email string, salt string) string {
 	hash := sha256.New()
-	hash.Write([]byte(salt + email))
+	_, _ = hash.Write([]byte(salt + email))
```
Comment (denialanderror) [inline middleware/jwt_auth.go:77]: Is this a Golang convention?

File: person/index_handler.go [hunk 3]
```diff
@@ -47,19 +64,11 @@ func (i IndexHandler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
 		resp.Results = append(resp.Results, *i.es.Index(p))
 	}
 
-	jsonResp, err := json.Marshal(resp)
-	if err != nil {
-		i.logger.Println(err.Error())
-		http.Error(rw, "Unable to encode response object to JSON", http.StatusBadRequest)
-		return
-	}
+	jsonResp, _ := json.Marshal(resp)
```
Comment (denialanderror) [inline person/index_handler.go:67]: Are we no longer interested in this error?

**Review by denialanderror — APPROVED** [review]
A few questions about ignoring errors and convention but it all looks fine to me. Don't know if you would also want a review from someone more familiar with Golang though.

File: middleware/jwt_auth.go (hunk 2 above)
Comment (devkokov) [reply middleware/jwt_auth.go:77]: Gosec throws an error without the `_, _` bit. I guess it forces you to explicitly ignore returned errors/values for better visibility

File: person/index_handler.go (hunk 3 above)
Comment (devkokov) [reply person/index_handler.go:67]: It's one of those errors that can never be reached, unless the structure of `resp` (`IndexResponse`) is not marshalable which would be a persistent error picked up by automated tests.

File: elasticsearch/client.go (hunk 1 above)
Comment (devkokov) [reply elasticsearch/client.go:76]: I was thinking that if signing fails the request should also fail with an appropriate error message, which is handled and logged further down 🤷  happy to revert it though

**Pull request merged**